Dockerfile-cli-only

```diff
@@ -0,0 +1,16 @@
+# syntax=docker/dockerfile:1.9@sha256:fe40cf4e92cd0c467be2cfc30657a680ae2398318afd50b0c80585784c604f28
+
+# Copyright Authors of Cilium
+# SPDX-License-Identifier: Apache-2.0
+
+FROM docker.io/library/golang:1.22.6-alpine3.19@sha256:1bad39361dd21f2f881ce10ff810e40e5be3eba89a0b61e762e05ec42f9bbaf2 AS builder
```

Comment:
Can we avoid allowing this to diverge over time from the CI builder?

Reply:
I think we can by building the CI image of this one.

So to make it clear, Dockerfile should build only cilium-cli and we should have another Dockerfile building cilium-cli-ci of that one so no diverging happens

Reply:
https://github.com/cilium/cilium-cli/pull/2770/commits/bddd9844b646277013e4b67f828e996209d4f30f

I think this resolves everything with a single dockerfile